pkg/controller/generic/transform/options.go

```diff
@@ -5,13 +5,16 @@
 package transform
 
 import (
+	"go.uber.org/zap"
+
 	"github.com/cosi-project/runtime/pkg/controller"
 	"github.com/cosi-project/runtime/pkg/state"
 )
 
 // ControllerOptions configures TransformController.
 type ControllerOptions struct {
 	extraEventCh            <-chan struct{}
+	onShutdownCallback      func(logger *zap.Logger)
```

Comment:
Maybe it makes sense to pass `controller.ReaderWriter` here?

Reply:
I don't care, whatever works better